### ISSUE TITLE
Implementation for other shader types

### ISSUE THREAD

**Issue by PlayerWithoutName** [storm-enroute/macrogl#43]
It would be good to have Geometry, Tessellation Control and Evaluation and Compute Shaders

**Comment by axel22** [comment]
Agreed. Note that geometry and compute shaders are [already available](https://github.com/storm-enroute/macrogl/blob/master/src/macroglex/scala/org/macrogl/ex/Program.scala) - I use compute shaders heavily in my projects.

About other shader types, I can't guarantee that I'll jump on it right away, given my bandwidth. If you feel like to submit a PR with the patch and an example program, feel free to go ahead.

**Comment by PlayerWithoutName** [comment]
How do i use geometry shader? And why is macrogl divided into non-ex and ex part?

**Comment by axel22** [comment]
The non-ex part is all the functionality that will work both with LWJGL and
WebGL (assuming the GPU supports it).

The ex part is functionality that WebGL does not support (typically, >
OpenGL 3.2, and some legacy things that WebGL does not support).

You can use the geometry shaders by instantiating a Program with the proper
Shader object:

      val pp = new Program("BasicTriangle")(
        Program.Shader.Vertex(vertexSource),
        ex.Program.Shader.Geometry(geometrySource),
        Program.Shader.Fragment(fragmentSource))

Note: for my purposes, I use the compute shaders, so I haven't tried the
geometry shaders out yet.
Ideally, we should have a working example with geometry shaders in the
examples programs.
